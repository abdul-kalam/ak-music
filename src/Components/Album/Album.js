import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import {
  AlbumContainer,
  AlbumItem,
  Details,
  DetailsContent,
  Title,
  Artist,
  AlbumImage,
  AlbumCover,
  Favorite,
} from './Album.style';
import { FavoriteUnSaved } from '../../Assets/FavoriteUnSaved';
import { FavoriteSaved } from '../../Assets/FavoriteSaved';
import { addFavorite, removeFavorite } from '../../Store/Reducers/musicSlice';
import {
  setLocalStorageValue,
  getLocalStorageValue,
} from '../../Modules/utils';

const FAVORITE_ALBUMS = 'favorite-albums';

export const Album = ({ album }) => {

  const [isFavorite, setIsFavorite] = useState(album.isFavorite || false);
  const dispatch = useDispatch();
  const history = useHistory();
  const id = album.id.attributes['im:id'];
  const title = album['im:name'].label;
  const img = album['im:image'][2]['label'];
  const artist = album['im:artist'].label;

  const handleOnClick = () =>
    history.push(`/album/${title.toLowerCase().split(' ').join('-')}/${id}`);

  const handleFavorite = (e) => {
    e.stopPropagation();
    setIsFavorite(!isFavorite);
    if (isFavorite) {
      let favorites = getLocalStorageValue(FAVORITE_ALBUMS) || [];
      favorites = favorites.filter(
        (album) => album.id.attributes['im:id'] !== id
      );
      setLocalStorageValue(FAVORITE_ALBUMS, favorites);
      dispatch(removeFavorite({ id: id }));
    } else {
      let favorites = getLocalStorageValue(FAVORITE_ALBUMS) || [];
      favorites.push({ ...album, ...{ isFavorite: true } });
      setLocalStorageValue(FAVORITE_ALBUMS, favorites);
      dispatch(addFavorite({ id: id }));
    }
  };

  return (
    <AlbumContainer onClick={handleOnClick} data-testid='album'>
      <AlbumItem>
        <AlbumImage>
          <AlbumCover alt={title} src={img} loading="lazy" />
        </AlbumImage>
        <Details>
          <DetailsContent>
            <Title>{title}</Title>
            <Artist>{artist}</Artist>
          </DetailsContent>
          <Favorite onClick={handleFavorite}>
            {isFavorite ? <FavoriteSaved /> : <FavoriteUnSaved />}
          </Favorite>
        </Details>
      </AlbumItem>
    </AlbumContainer>
  );
};
