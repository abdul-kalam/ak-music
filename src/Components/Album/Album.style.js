import styled from 'styled-components'

export const AlbumContainer = styled.li`
  width: 206px;
  margin: 10px;
  cursor: pointer;
  user-select: none;
`
export const AlbumItem = styled.div`
  width: 100%;
  height: 100%;
  &:hover {
    transform: scale(1.1);
  }
  transition: transform 0.1s ease-in;
`

export const AlbumImage = styled.div`
  margin: 0;
  position: relative;
  display: block;
  border-radius: 6px;
  box-shadow: 0 4px 14px rgb(0 0 0 / 10%);
  width: 100%;
`

export const Details = styled.div`
  height: 50px;
  padding-top: 6px;
  word-break: break-word;
  display: flex;
  font-size: 12px;
  line-height: 1.25;
  font-weight: 400;
  letter-spacing: 0;
  color: rgba(255, 255, 255, 0.55);
`

export const DetailsContent = styled.div`
  flex: 1;
`
export const Title = styled.span`
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
  white-space: normal;

  text-decoration: none !important;
  color: rgba(255, 255, 255, 0.85);
`

export const Artist = styled.div`
  flex: 1;
`

export const AlbumCover = styled.img`
  margin: 0;
  display: block;
  border-radius: 6px;
  width: 206px;
  opacity: 1;
  &:hover {
    opacity: 0.85;
  }
  transition: opacity 0.1s ease-in;
`

export const Favorite = styled.div`

`;

