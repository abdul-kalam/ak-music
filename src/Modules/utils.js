export const getLocalStorageValue = (key) => {
  const value = localStorage.getItem(key);
  if (value) {
    return JSON.parse(value);
  }
  return '';
};

export const setLocalStorageValue = (key, value) => {
  value = JSON.stringify(value);
  localStorage.setItem(key, value);
};
