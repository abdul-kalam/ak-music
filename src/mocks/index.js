export const mockTopAlbum = {
  feed: {
    author: {
      name: {
        label: 'iTunes Store',
      },
      uri: {
        label: 'http://www.apple.com/itunes/',
      },
    },
    entry: [
      {
        isFavorite: true,
        'im:name': {
          label: 'Welcome 2 America',
        },
        'im:image': [
          {
            label:
              'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/55/46/19/55461993-6fd7-d703-9b39-03043f585203/886449187249.jpg/55x55bb.png',
            attributes: {
              height: '55',
            },
          },
          {
            label:
              'https://is3-ssl.mzstatic.com/image/thumb/Music114/v4/55/46/19/55461993-6fd7-d703-9b39-03043f585203/886449187249.jpg/60x60bb.png',
            attributes: {
              height: '60',
            },
          },
          {
            label:
              'https://is1-ssl.mzstatic.com/image/thumb/Music114/v4/55/46/19/55461993-6fd7-d703-9b39-03043f585203/886449187249.jpg/170x170bb.png',
            attributes: {
              height: '170',
            },
          },
        ],
        'im:itemCount': {
          label: '12',
        },
        'im:price': {
          label: '$9.99',
          attributes: {
            amount: '9.99',
            currency: 'USD',
          },
        },
        'im:contentType': {
          'im:contentType': {
            attributes: {
              term: 'Album',
              label: 'Album',
            },
          },
          attributes: {
            term: 'Music',
            label: 'Music',
          },
        },
        rights: {
          label:
            '℗ 2021 NPG Records, Inc. under exclusive license to Legacy Recordings',
        },
        title: {
          label: 'Welcome 2 America - Prince',
        },
        link: {
          attributes: {
            rel: 'alternate',
            type: 'text/html',
            href: 'https://music.apple.com/us/album/welcome-2-america/1561643530?uo=2',
          },
        },
        id: {
          label:
            'https://music.apple.com/us/album/welcome-2-america/1561643530?uo=2',
          attributes: {
            'im:id': '1561643530',
          },
        },
        'im:artist': {
          label: 'Prince',
          attributes: {
            href: 'https://music.apple.com/us/artist/prince/155814?uo=2',
          },
        },
        category: {
          attributes: {
            'im:id': '15',
            term: 'R&B/Soul',
            scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
            label: 'R&B/Soul',
          },
        },
        'im:releaseDate': {
          label: '2021-07-30T00:00:00-07:00',
          attributes: {
            label: 'July 30, 2021',
          },
        },
      },
      {
        isFavorite: true,
        'im:name': {
          label: 'Into The Mystery',
        },
        'im:image': [
          {
            label:
              'https://is2-ssl.mzstatic.com/image/thumb/Music125/v4/f5/48/5f/f5485f4c-458e-425e-fa60-e3ce2871cb00/075679788856.jpg/55x55bb.png',
            attributes: {
              height: '55',
            },
          },
          {
            label:
              'https://is4-ssl.mzstatic.com/image/thumb/Music125/v4/f5/48/5f/f5485f4c-458e-425e-fa60-e3ce2871cb00/075679788856.jpg/60x60bb.png',
            attributes: {
              height: '60',
            },
          },
          {
            label:
              'https://is2-ssl.mzstatic.com/image/thumb/Music125/v4/f5/48/5f/f5485f4c-458e-425e-fa60-e3ce2871cb00/075679788856.jpg/170x170bb.png',
            attributes: {
              height: '170',
            },
          },
        ],
        'im:itemCount': {
          label: '12',
        },
        'im:price': {
          label: '$7.99',
          attributes: {
            amount: '7.99',
            currency: 'USD',
          },
        },
        'im:contentType': {
          'im:contentType': {
            attributes: {
              term: 'Album',
              label: 'Album',
            },
          },
          attributes: {
            term: 'Music',
            label: 'Music',
          },
        },
        rights: {
          label: '℗ 2021 Elektra Records LLC',
        },
        title: {
          label: 'Into The Mystery - NEEDTOBREATHE',
        },
        link: {
          attributes: {
            rel: 'alternate',
            type: 'text/html',
            href: 'https://music.apple.com/us/album/into-the-mystery/1566670096?uo=2',
          },
        },
        id: {
          label:
            'https://music.apple.com/us/album/into-the-mystery/1566670096?uo=2',
          attributes: {
            'im:id': '1566670096',
          },
        },
        'im:artist': {
          label: 'NEEDTOBREATHE',
          attributes: {
            href: 'https://music.apple.com/us/artist/needtobreathe/113247214?uo=2',
          },
        },
        category: {
          attributes: {
            'im:id': '21',
            term: 'Rock',
            scheme: 'https://music.apple.com/us/genre/music-rock/id21?uo=2',
            label: 'Rock',
          },
        },
        'im:releaseDate': {
          label: '2021-07-30T00:00:00-07:00',
          attributes: {
            label: 'July 30, 2021',
          },
        },
      },
      {
        'im:name': {
          label: 'The House Is Burning',
        },
        'im:image': [
          {
            label:
              'https://is2-ssl.mzstatic.com/image/thumb/Music115/v4/3f/28/c8/3f28c8e9-c28a-1087-8ccb-654c35bdfbb8/093624878070.jpg/55x55bb.png',
            attributes: {
              height: '55',
            },
          },
          {
            label:
              'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/3f/28/c8/3f28c8e9-c28a-1087-8ccb-654c35bdfbb8/093624878070.jpg/60x60bb.png',
            attributes: {
              height: '60',
            },
          },
          {
            label:
              'https://is3-ssl.mzstatic.com/image/thumb/Music115/v4/3f/28/c8/3f28c8e9-c28a-1087-8ccb-654c35bdfbb8/093624878070.jpg/170x170bb.png',
            attributes: {
              height: '170',
            },
          },
        ],
        'im:itemCount': {
          label: '16',
        },
        'im:price': {
          label: '$9.99',
          attributes: {
            amount: '9.99',
            currency: 'USD',
          },
        },
        'im:contentType': {
          'im:contentType': {
            attributes: {
              term: 'Album',
              label: 'Album',
            },
          },
          attributes: {
            term: 'Music',
            label: 'Music',
          },
        },
        rights: {
          label:
            '℗ 2021 Top Dawg Entertainment, under exclusive license to Warner Records Inc.',
        },
        title: {
          label: 'The House Is Burning - Isaiah Rashad',
        },
        link: {
          attributes: {
            rel: 'alternate',
            type: 'text/html',
            href: 'https://music.apple.com/us/album/the-house-is-burning/1577158260?uo=2',
          },
        },
        id: {
          label:
            'https://music.apple.com/us/album/the-house-is-burning/1577158260?uo=2',
          attributes: {
            'im:id': '1577158260',
          },
        },
        'im:artist': {
          label: 'Isaiah Rashad',
          attributes: {
            href: 'https://music.apple.com/us/artist/isaiah-rashad/605391263?uo=2',
          },
        },
        category: {
          attributes: {
            'im:id': '18',
            term: 'Hip Hop/Rap',
            scheme:
              'https://music.apple.com/us/genre/music-hip-hop-rap/id18?uo=2',
            label: 'Hip-Hop/Rap',
          },
        },
        'im:releaseDate': {
          label: '2021-07-30T00:00:00-07:00',
          attributes: {
            label: 'July 30, 2021',
          },
        },
      },
    ],
    updated: {
      label: '2021-08-04T13:45:32-07:00',
    },
    rights: {
      label: 'Copyright 2008 Apple Inc.',
    },
    title: {
      label: 'iTunes Store: Top Albums',
    },
    icon: {
      label: 'http://itunes.apple.com/favicon.ico',
    },
    link: [
      {
        attributes: {
          rel: 'alternate',
          type: 'text/html',
          href: 'https://itunes.apple.com/WebObjects/MZStore.woa/wa/viewTop?cc=us&id=1&popId=11',
        },
      },
      {
        attributes: {
          rel: 'self',
          href: 'https://mzstoreservices-int-st.itunes.apple.com/us/rss/topalbums/limit=100/json',
        },
      },
    ],
    id: {
      label:
        'https://mzstoreservices-int-st.itunes.apple.com/us/rss/topalbums/limit=100/json',
    },
  },
  titleListMock: [
    {
      id: '1566670096',
      name: 'Welcome to America',
      image: 'www.test.com/img.jpg',
    },
    {
      id: '1566670094',
      name: 'Welcome to India',
      image: 'www.test.com/img1.jpg',
    },
  ],
};

export const mockAlbumSongs = {
  resultCount: 4,
  results: [
    {
      wrapperType: 'collection',
      collectionType: 'Album',
      artistId: 4779205,
      collectionId: 1570554138,
      amgArtistId: 825908,
      artistName: 'Chris Young',
      collectionName: 'Famous Friends',
      collectionCensoredName: 'Famous Friends',
      artistViewUrl:
        'https://music.apple.com/us/artist/chris-young/4779205?uo=4',
      collectionViewUrl:
        'https://music.apple.com/us/album/famous-friends/1570554138?uo=4',
      artworkUrl60:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/60x60bb.jpg',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/100x100bb.jpg',
      collectionPrice: 11.99,
      collectionExplicitness: 'notExplicit',
      trackCount: 14,
      copyright: '℗ 2021 Sony Music Entertainment. All rights reserved.',
      country: 'USA',
      currency: 'USD',
      releaseDate: '2021-08-06T07:00:00Z',
      primaryGenreName: 'Country',
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 4779205,
      collectionId: 1570554138,
      trackId: 1570554142,
      artistName: 'Chris Young',
      collectionName: 'Famous Friends',
      trackName: 'Raised on Country',
      collectionCensoredName: 'Famous Friends',
      trackCensoredName: 'Raised on Country',
      artistViewUrl:
        'https://music.apple.com/us/artist/chris-young/4779205?uo=4',
      collectionViewUrl:
        'https://music.apple.com/us/album/raised-on-country/1570554138?i=1570554142&uo=4',
      trackViewUrl:
        'https://music.apple.com/us/album/raised-on-country/1570554138?i=1570554142&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/10/c0/3f/10c03f67-bb34-ff6b-403b-4c6070fb56ad/mzaf_15785209836383516861.plus.aac.p.m4a',
      artworkUrl30:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/30x30bb.jpg',
      artworkUrl60:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/60x60bb.jpg',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/100x100bb.jpg',
      collectionPrice: 11.99,
      trackPrice: 1.29,
      releaseDate: '2019-01-04T08:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 14,
      trackNumber: 1,
      trackTimeMillis: 176520,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Country',
      isStreamable: true,
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 4779205,
      collectionId: 1570554138,
      trackId: 1570554143,
      artistName: 'Chris Young & Kane Brown',
      collectionName: 'Famous Friends',
      trackName: 'Famous Friends',
      collectionCensoredName: 'Famous Friends',
      trackCensoredName: 'Famous Friends',
      collectionArtistName: 'Chris Young',
      artistViewUrl:
        'https://music.apple.com/us/artist/chris-young/4779205?uo=4',
      collectionViewUrl:
        'https://music.apple.com/us/album/famous-friends/1570554138?i=1570554143&uo=4',
      trackViewUrl:
        'https://music.apple.com/us/album/famous-friends/1570554138?i=1570554143&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/42/7a/5f/427a5f4b-5e18-fdb6-3ba8-56a2d06483b0/mzaf_2809262009335345042.plus.aac.p.m4a',
      artworkUrl30:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/30x30bb.jpg',
      artworkUrl60:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/60x60bb.jpg',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/100x100bb.jpg',
      collectionPrice: 11.99,
      trackPrice: 1.29,
      releaseDate: '2020-11-20T08:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 14,
      trackNumber: 2,
      trackTimeMillis: 166507,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Country',
      isStreamable: true,
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 4779205,
      collectionId: 1570554138,
      trackId: 1570554144,
      artistName: 'Chris Young & Lauren Alaina',
      collectionName: 'Famous Friends',
      trackName: "Town Ain't Big Enough",
      collectionCensoredName: 'Famous Friends',
      trackCensoredName: "Town Ain't Big Enough",
      collectionArtistName: 'Chris Young',
      artistViewUrl:
        'https://music.apple.com/us/artist/chris-young/4779205?uo=4',
      collectionViewUrl:
        'https://music.apple.com/us/album/town-aint-big-enough/1570554138?i=1570554144&uo=4',
      trackViewUrl:
        'https://music.apple.com/us/album/town-aint-big-enough/1570554138?i=1570554144&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/d3/0d/69/d30d693f-570d-0bc0-3c70-7c7a272baec3/mzaf_7645675846915846385.plus.aac.p.m4a',
      artworkUrl30:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/30x30bb.jpg',
      artworkUrl60:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/60x60bb.jpg',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/d1/6e/a9/d16ea9a6-c2a4-90aa-99e7-a981ee8cb689/source/100x100bb.jpg',
      collectionPrice: 11.99,
      trackPrice: 1.29,
      releaseDate: '2019-07-12T07:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 14,
      trackNumber: 3,
      trackTimeMillis: 171800,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Country',
      isStreamable: true,
    },
  ],
};
