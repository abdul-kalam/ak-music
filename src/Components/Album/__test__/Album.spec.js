import React from 'react';
import { render, screen } from '../../../Modules/test-utils';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { Album } from './../Album';
import { mockTopAlbum} from './../../../mocks';
import userEvent from '@testing-library/user-event';



describe('Album component', () => {
  test('component should render with passed data', () => {
    const mockAlbum = mockTopAlbum.feed.entry[0];

    render(<Album album={mockAlbum} />);

    expect(screen.getAllByAltText('Welcome 2 America')).toHaveLength(1);
  });

  test('component should navigate to album details page on clicking', () => {
    const mockAlbum = mockTopAlbum.feed.entry[0];
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <Album album={mockAlbum} />)
      </Router>
    );

    const album = screen.getByText('Welcome 2 America');
    expect(album).toBeInTheDocument();
    userEvent.click(album);
    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe(
      '/album/welcome-2-america/1561643530'
    );
  });

  test('component should toggle favorite state of album on clicking', () => {
    const mockAlbum = mockTopAlbum.feed.entry[0];

    const { asFragment } = render(<Album album={mockAlbum} />);

    const favorite = screen.getAllByTitle('Favorite');
    expect(favorite[0]).toBeInTheDocument();
    userEvent.click(favorite[0]);
    const unfavorite = screen.getByTitle('UnFavorite');
    expect(unfavorite).toBeInTheDocument();
    expect(favorite[0]).not.toBeInTheDocument();
    userEvent.click(unfavorite);
    expect(screen.getAllByTitle('Favorite')[0]).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
