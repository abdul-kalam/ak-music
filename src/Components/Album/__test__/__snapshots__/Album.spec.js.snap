// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component component should toggle favorite state of album on clicking 1`] = `
<DocumentFragment>
  <li
    class="sc-bdnxRM drQndn"
    data-testid="album"
  >
    <div
      class="sc-gtsrHT cLORvp"
    >
      <div
        class="sc-dlnjwi bWffnD"
      >
        <img
          alt="Welcome 2 America"
          class="sc-iCoGMd jNGNJS"
          loading="lazy"
          src="https://is1-ssl.mzstatic.com/image/thumb/Music114/v4/55/46/19/55461993-6fd7-d703-9b39-03043f585203/886449187249.jpg/170x170bb.png"
        />
      </div>
      <div
        class="sc-hKFxyN lgKhgn"
      >
        <div
          class="sc-eCApnc hPTdnV"
        >
          <span
            class="sc-jSFjdj dprczd"
          >
            Welcome 2 America
          </span>
          <div
            class="sc-gKAaRy dGeeFs"
          >
            Prince
          </div>
        </div>
        <div
          class="sc-fujyAs cNlUce"
        >
          <svg
            height="30px"
            id="Layer_1"
            version="1.1"
            viewBox="0 0 455 455"
            width="30px"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <title>
              Favorite
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
              stroke="none"
              stroke-width="2"
            >
              <g
                fill="#fa2d48"
                fill-rule="nonzero"
              >
                <path
                  d="M326.632,10.346c-38.733,0-74.991,17.537-99.132,46.92c-24.141-29.383-60.399-46.92-99.132-46.92 C57.586,10.346,0,67.931,0,138.714c0,55.426,33.049,119.535,98.23,190.546c50.162,54.649,104.729,96.96,120.257,108.626l9.01,6.769 l9.009-6.768c15.53-11.667,70.099-53.979,120.26-108.625C421.95,258.251,455,194.141,455,138.714 C455,67.931,397.414,10.346,326.632,10.346z"
                />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
                <g />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
